nautobot/core/jobs/__init__.py

```diff
@@ -347,5 +366,168 @@ def run(self, *, content_type, csv_data=None, csv_file=None, roll_back_if_error=
             raise RunJobTaskFailed("CSV import not fully successful, see logs")
 
 
-jobs = [ExportObjectList, GitRepositorySync, GitRepositoryDryRun, ImportObjects, LogsCleanup, RefreshDynamicGroupCaches]
+class BulkEditObjects(Job):
+    """System Job to bulk Edit objects."""
+
+    content_type = ObjectVar(
+        model=ContentType,
+        description="Type of objects to import",
+        query_params={"has_serializer": True},
+        # query_params={"has_serializer": True, "has_bulk_edit_form": True, "has_filter_form": True},
+    )
+    post_data = JSONVar(description="Data to update with i.e request.POST")
+    edit_all = BooleanVar(description="Bulk Edit all object / all filtered objects", required=False)
+    filter_query_params = JSONVar(label="Filter Query Params", required=False)
+
+    class Meta:
+        name = "Bulk Edit Objects"
+        description = "Bulk edit objects."
+        has_sensitive_variables = False
+        soft_time_limit = 1800
+        time_limit = 2000
+
+    def _update_objects(self, model, form, filter_query_params, edit_all, nullified_fields):
+        with deferred_change_logging_for_bulk_operation():
+            updated_objects = []
+            filterset_cls = get_filterset_for_model(model)
+            if edit_all:
+                if filterset_cls:
+                    queryset = filterset_cls(filter_query_params).qs.restrict(self.user, "edit")
+                else:
+                    queryset = model.objects.restrict(self.user, "edit")
+            else:
+                queryset = model.objects.restrict(self.user, "edit").filter(pk__in=form.cleaned_data["pk"])
+
+            form_custom_fields = getattr(form, "custom_fields", [])
+            form_relationships = getattr(form, "relationships", [])
+            standard_fields = [
+                field
+                for field in form.fields
+                if field not in form_custom_fields + form_relationships + ["pk"] + ["object_note"]
+            ]
+
+            for obj in queryset:
```

Comment:
Just a general note - let's make sure the job has sufficient logging, especially of progress for long-running operations.